### PR TITLE
Webhook migrate script

### PR DESCRIPTION
**Pull request by manishtomar**
First task of #1039. This creates `CQLQueryExecute` intent along with its performer and dispatcher to migrate the webhooks. Would like to know if having separate `otter.effect_dispatcher.get_cql_dispatcher` is a good idea @radix ?

I've tested this from dev vm by copying data from SYD prod.

### CODE REVIEW

File: otter/effect_dispatcher.py [hunk 1]
```diff
@@ -32,7 +36,7 @@ def get_simple_dispatcher(reactor):
             Authenticate: perform_authenticate,
             InvalidateToken: perform_invalidate_token,
             Request: perform_request,
-            Retry: perform_retry,
+            Retry: perform_retry
```
Comment (radix) [inline otter/effect_dispatcher.py]: That trailing comma was intentional (exactly so diffs would be simpler), could you revert this change?

**Comment by radix** [comment]
get_cql_dispatcher seems fine to me :)

File: otter/test/models/test_cass_models.py [hunk 2]
```diff
@@ -106,6 +114,28 @@ def _cassandrify_data(list_of_dicts):
...
+    """
+    Tests for :func:`perform_cql_query` function
+    """
+
+    def test_perform_query_sync(self):
+        """
+        Calls given connection's execute
+        """
+        conn = mock.Mock(spec=CQLClient)
+        conn.execute.return_value = 'ret'
```
Comment (radix) [inline otter/test/models/test_cass_models.py]: Could you change this to be a defer.succeed('ret')? this test could succeed without the @deferred_performer decorator, but changing the return value to succeed('ret') should make it test that it's really dealing with deferreds correctly.

File: otter/models/cass.py [hunk 3]
```diff
@@ -1486,6 +1510,46 @@ def _do_webhook_lookup(webhook_rec):
...
+                                params={},
+                                consistency_level=ConsistencyLevel.ONE))])
+        return eff.on(
+            lambda (whooks, wkeys): set(freeze(whooks)) - set(freeze(wkeys)))
+
+    def add_webhook_keys(self, webhook_keys):
+        """
+        Add webhook keys to webhook_keys table
+
+        :return: Effect of CQLQueryExecute
```
Comment (radix) [inline otter/models/cass.py]: The "Effect of *" idiom should generally be used to describe the *result* of the Effect, not the intent of the effect. So, in this case it'd be an Effect of... whatever connection.execute() will return in this case. (None?)

File: otter/models/cass.py [hunk 4]
```diff
@@ -1486,6 +1510,46 @@ def _do_webhook_lookup(webhook_rec):
         d.addCallback(_do_webhook_lookup)
         return d
 
+    def get_webhook_index_only(self):
+        """
+        Get webhook info that is there in webhook index but is not there in
+        webhook_keys table
+
+        :return: Effect of CQLQueryExecute
```
Comment (radix) [inline otter/models/cass.py]: Same `Effect of *` comment here.

File: otter/test/models/test_cass_models.py [hunk 5]
```diff
@@ -2527,6 +2557,68 @@ def test_add_scaling_policy_cron(self):
...
+
+    def setUp(self):
+        """
+        Sample CassScalingGroup object
+        """
+        self.store = CassScalingGroupCollection(None, None)
+
+    def test_webhook_index_only(self):
+        """
+        Test `get_webhook_index_only`
```
Comment (glyph) [inline otter/test/models/test_cass_models.py]: What aspect of `get_webhook_index_only` is this testing?  `"w2": "2"` is not the most self-documenting of examples :).

Comment (manishtomar) [inline otter/test/models/test_cass_models.py]: Well.. I would just have to repeat the docstring of `get_webhook_index_only` here to make sense. Do you think its ok to put `Refer get_webhook_index_only docs` here?

Comment (glyph) [inline otter/test/models/test_cass_models.py]: I prefer to avoid that; the test is always testing something more nuanced than the public documentation.  In this case, one obvious distinction is that `get_webhook_index_only`'s docstring does not note that the effects take place in parallel, yet this test cares and will fail if the effect's intent is sequential rather than parallel.

Comment (manishtomar) [inline otter/test/models/test_cass_models.py]: Hmm.. OK. I didn't think from that aspect. I'll change the docstring.

File: otter/test/models/test_cass_models.py [hunk 6]
```diff
@@ -106,6 +114,28 @@ def _cassandrify_data(list_of_dicts):
     return _de_identify(list_of_dicts)
 
 
+class PerformTests(SynchronousTestCase):
+    """
+    Tests for :func:`perform_cql_query` function
+    """
+
+    def test_perform_cql_query(self):
```
Comment (cyli) [inline otter/test/models/test_cass_models.py]: +1 for actually testing perform :D

**Comment by cyli** [comment]
I feel like I'm somewhat shaky on effect, so I don't feel super comfortable reviewing the dispatcher + tests, but the rest of it LGTM.  @radix maybe?

My non-blocking concern is I'm not sure how many webhooks are there in the webhooks table - I dunno if grabbing all of it at once would cause any problems.

If you've already tested against our biggest DCs then it it's fine as is, otherwise maybe we can do this on a per-tenant basis?  (Although that can be done in a different PR).

**Comment by radix** [comment]
The effect/dispatcher/performer stuff looks fine to me. I'm +1 on merging once cyli's concern about data size is addressed.

**Comment by manishtomar** [comment]
> My non-blocking concern is I'm not sure how many webhooks are there in the webhooks table

I've done that against IAD and ORD (AFAIR). So, I think we are good in that sense.
